Readme.md

```diff
@@ -45,6 +45,8 @@ $ composer update
 
 ### Laravel 5.x
 
+For Laravel 5.5 you can skip step 1 and 2 and let auto-discovery feature do it´s magic.
```

Comment:
I'd just remove the extra steps and assume most people are using 5.5 from now on.

Reply:
Without any reference of adding the service provider to config/app.php??

What about making something like this:

### Laravel 5.5

**1-** Publish the configuration file

```bash
$ php artisan vendor:publish --provider="Folklore\GraphQL\ServiceProvider"
```

**2-** Review the configuration file

```
config/graphql.php
```

### Laravel <= 5.4.x

**1-** Add the service provider to your `app/config/app.php` file
```php
Folklore\GraphQL\ServiceProvider::class,
```

**2-** Add the facade to your `app/config/app.php` file
```php
'GraphQL' => Folklore\GraphQL\Support\Facades\GraphQL::class,
```

**3-** Publish the configuration file

```bash
$ php artisan vendor:publish --provider="Folklore\GraphQL\ServiceProvider"
```

**4-** Review the configuration file

```
config/graphql.php
```


### Lumen

**1-** Load the service provider in `bootstrap/app.php`
```php
$app->register(Folklore\GraphQL\LumenServiceProvider::class);
```

**2-** For using the facade you have to uncomment the line `$app->withFacades();` in `bootstrap/app.php`

After uncommenting this line you have the `GraphQL` facade enabled

```php
$app->withFacades();
```

**3-** Publish the configuration file

```bash
$ php artisan graphql:publish
```

**4-** Load configuration file in `bootstrap/app.php`

*Important*: this command needs to be executed before the registration of the service provider

```php
$app->configure('graphql');
...
$app->register(Folklore\GraphQL\LumenServiceProvider::class)
```

**5-** Review the configuration file

```
config/graphql.php
```

Reply:
> Without any reference of adding the service provider to config/app.php??

Yes because it's not needed with auto-discovery.

I like your solution from above 👍